apps/hermes/server/src/api/rest/v2/sse.rs

```diff
@@ -93,7 +96,14 @@ where
     // Convert the broadcast receiver into a Stream
     let stream = BroadcastStream::new(update_rx);
 
+    // Set connection deadline
+    let connection_deadline = Instant::now() + MAX_CONNECTION_DURATION;
+
     let sse_stream = stream
+        .take_while(move |_| {
+            let now = Instant::now();
+            now < connection_deadline
```

Comment:
I recommend using `start_time = Instant::now()` and then check `start_time.elapsed() < MAX_....`